cfme/tests/infrastructure/test_timelines.py

```diff
@@ -1,242 +1,128 @@
 from cfme.utils.log import logger
-from cfme.utils.providers import ProviderFilter
 from cfme.utils.wait import wait_for
-from fixtures.provider import setup_one_or_skip
-
 
-pytestmark = [pytest.mark.tier(2)]
+pytestmark = [pytest.mark.tier(2), pytest.mark.usefixtures("setup_provider_modscope")]
+pytest_generate_tests = testgen.generate([InfraProvider], scope='module')
 
 
 @pytest.fixture(scope='module')
-def a_provider(request):
-    BaseProvider.clear_providers()
-    not_scvmm = ProviderFilter(classes=[SCVMMProvider],
-                               inverted=True)  # scvmm doesn't provide events
-    all_prov = ProviderFilter(classes=[InfraProvider])
-    return setup_one_or_skip(request, filters=[not_scvmm, all_prov])
-
-
-@pytest.fixture(scope="module")
-def new_vm(request, a_provider):
-    vm = VM.factory(random_vm_name("timelines", max_length=16), a_provider)
-
-    request.addfinalizer(vm.delete_from_provider)
-
-    if not a_provider.mgmt.does_vm_exist(vm.name):
-        logger.info("deploying %s on provider %s", vm.name, a_provider.key)
-        vm.create_on_provider(allow_skip="default", find_in_cfme=True)
-    return vm
-
-
-@pytest.yield_fixture(scope="module")
-def mark_vm_as_appliance(new_vm, appliance):
-    # set diagnostics vm
-    relations_view = navigate_to(new_vm, 'EditManagementEngineRelationship')
-    server_name = "{name} ({sid})".format(name=appliance.server.name, sid=appliance.server.sid)
-    relations_view.form.server.select_by_visible_text(server_name)
-    relations_view.form.save_button.click()
-    yield
-    # unset diagnostics vm
-    relations_view = navigate_to(new_vm, 'EditManagementEngineRelationship')
-    relations_view.form.server.select_by_visible_text('<Not a Server>')
-    relations_view.form.save_button.click()
-
-
-@pytest.fixture(scope="module")
-def gen_events(new_vm):
-    logger.debug('Starting, stopping VM')
-    mgmt = new_vm.provider.mgmt
-    mgmt.stop_vm(new_vm.name)
-    mgmt.start_vm(new_vm.name)
-
-
-def count_events(target, vm):
-    timelines_view = navigate_to(target, 'Timelines')
-    if isinstance(target, Server):
-        timelines_view = timelines_view.timelines
-    timelines_view.filter.time_position.select_by_visible_text('centered')
-    timelines_view.filter.apply.click()
-    found_events = []
-    for evt in timelines_view.chart.get_events():
-        if not hasattr(evt, 'source_vm'):
-            # BZ(1428797)
-            logger.warn("event {evt} doesn't have source_vm field. Probably issue".format(evt=evt))
-            continue
-        elif evt.source_vm == vm.name:
-            found_events.append(evt)
-
-    logger.info("found events: {evt}".format(evt="\n".join([repr(e) for e in found_events])))
-    return len(found_events)
-
-
-def test_infra_provider_event(gen_events, new_vm):
-    """Tests provider event on timelines
-
-    Metadata:
-        test_flag: timelines, provision
-    """
+def new_vm(provider):
+    logger.info('new_vm set up!')
```

Comment:
REQUIRED: if you decided to add more logging, please add it everywhere. In addition, please add necessary details to logging statements like vm name, etc.